internal/source/git_test.go

```diff
@@ -0,0 +1,329 @@
+package source
+
+import (
+	"bytes"
+	"context"
+	"os"
+
+	"github.com/go-git/go-git/v5/plumbing"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	kscheme "k8s.io/client-go/kubernetes/scheme"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+
+	rukpakv1alpha1 "github.com/operator-framework/rukpak/api/v1alpha1"
+)
+
+var _ = Describe("git.Unpack", func() {
+	It("should fail if bundle source type is not git", func() {
+		cfg := ctrl.GetConfigOrDie()
+		mgr, err := ctrl.NewManager(cfg, ctrl.Options{})
+		Expect(err).To(BeNil())
+		git := Git{mgr.GetClient(), "fakenamespace"}
+		bundle := &rukpakv1alpha1.Bundle{
+			Spec: rukpakv1alpha1.BundleSpec{
+				Source: rukpakv1alpha1.BundleSource{
+					Type: "faketype",
+				},
+			},
+		}
+		_, err = git.Unpack(context.TODO(), bundle)
```

Comment:
nit: we should use context.Background for tests to remain consistent with the rest of the codebase.